build-tools/packages/build-tools/src/fluidBuild/tasks/taskFactory.ts

```diff
@@ -95,6 +95,22 @@ function getTaskForExecutable(executable: string, context: BuildContext): TaskHa
 	return builtInHandler ?? UnknownLeafTask;
 }
 
+function getRunScriptName(command: string, packageManager: string): string | undefined {
+	// Remove the package manager name from the command
+	if (command.startsWith("npm run ")) {
+		return command.substring("npm run ".length);
+	}
+	// Only support yarn and pnpm for now
+	if (packageManager === "yarn" || packageManager === "pnpm") {
+		const packageManagerRun = `${packageManager} run `;
+
+		if (command.startsWith(packageManagerRun)) {
+			return command.substring(packageManagerRun.length);
+		}
+	}
+	return undefined;
+}
```

Comment:
I suggest having a list of supported package managers, including npm - we may have such a list already - and then doing something like this:

```suggestion
function getRunScriptName(command: string, packageManager: string): string | undefined {
	if (supportedPackageManagers.includes(packageManager)) {
		const packageManagerRun = `${packageManager} run `;

		// Remove the package manager name from the command
		if (command.startsWith(packageManagerRun)) {
			return command.substring(packageManagerRun.length);
		}
	}
	return undefined;
}
```

Reply:
well... I can create a list, but its use will be limited to "getRunScriptName" (and not just a list of supported package manager in fluid-build) as other package manager might not have the same "run" command.  Is that useful?

Reply:
I like the list to keep it all in one place, but I see your point. Leave as-is and we can clean up if/when we integrate build-infra into fluid-build.